### PR TITLE
flashbangs process flash/bang separately based on blindness/hearing and uses viewers(), which means xray users suffer too if they see them go off

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

tin

## Why It's Good For The Game

a fun way to nerf x-ray and make flashbangs more consistent

## Changelog
:cl:
balance: flashbangs process light/sound separately and uses viewers(), so xray users beware.
/:cl:

### CODE REVIEW

File: code/game/objects/items/grenades/flashbang.dm [hunk 1]
```diff
@@ -14,23 +14,34 @@
 	do_sparks(rand(5, 9), FALSE, src)
 	playsound(flashbang_turf, 'sound/weapons/flashbang.ogg', 100, TRUE, 8, 0.9)
 	new /obj/effect/dummy/lighting_obj (flashbang_turf, LIGHT_COLOR_WHITE, (flashbang_range + 2), 4, 2)
+	flashbang_mobs(flashbang_turf, flashbang_range)
 	for(var/mob/living/M in get_hearers_in_view(flashbang_range, flashbang_turf))
 		bang(get_turf(M), M)
```
Comment (Ghommie) [inline code/game/objects/items/grenades/flashbang.dm]: I suggest you to remove this two lines unless you also intended to double the pain for those poor tools caught in its AoE.

File: code/game/objects/items/grenades/flashbang.dm [hunk 2]
```diff
@@ -14,23 +14,34 @@
...
 		M.Knockdown(200)
 		M.soundbang_act(1, 200, 10, 15)
-
 	else
 		M.soundbang_act(1, max(200/max(1,distance), 60), rand(0, 5))
+
+/obj/item/grenade/flashbang/proc/flash(mob/living/M, turf/source)
+	if(M.stat == DEAD)	//They're dead!
+		return
+	var/distance = get_dist(get_turf(M, source))
```
Comment (Ghommie) [inline code/game/objects/items/grenades/flashbang.dm]:
```suggestion
	var/distance = get_dist(get_turf(M), source)
```

**Review by Ghommie — COMMENTED** [review]
All good save for the following...

File: code/game/objects/items/grenades/flashbang.dm (hunk 1 above)
Comment (silicons) [reply code/game/objects/items/grenades/flashbang.dm]: FUCK HOW DID I NOT NOTICE LMAO

**Review by Ghommie — APPROVED** [review]

**Pull request merged**